runelite-client/src/main/java/net/runelite/client/plugins/itemstats/potions/MoonlightPotion.java

```diff
@@ -0,0 +1,138 @@
+import static net.runelite.client.plugins.itemstats.stats.Stats.HERBLORE;
+import static net.runelite.client.plugins.itemstats.stats.Stats.PRAYER;
+import static net.runelite.client.plugins.itemstats.stats.Stats.STRENGTH;
+
+@RequiredArgsConstructor
+public class MoonlightPotion implements Effect
+{
+	private static final Stat[] OFFENSIVE_MELEE_STATS = {
+			ATTACK, STRENGTH
+	};
+
+	final StatBoost prayerRestore = new StatBoost(PRAYER, false)
+	{
+		@Override
+		public int heals(Client client)
+		{
+			int pray = PRAYER.getMaximum(client);
+			int herb = HERBLORE.getValue(client);
+			return herb < 38 ? 0 : (int) Math.max(pray * 0.25, herb * 0.3) + 7;
+		}
+	};
+
+	@Override
+	public StatsChanges calculate(Client client)
+	{
+		StatsChanges changes = new StatsChanges(0);
+		StatChange restoreChange = prayerRestore.effect(client);
+
+		SimpleStatBoost attackBoost = new SimpleStatBoost(ATTACK, true, getMeleeEffect(client, ATTACK));
+		SimpleStatBoost defenceBoost = new SimpleStatBoost(DEFENCE, true, getMeleeEffect(client, DEFENCE));
+		SimpleStatBoost strengthBoost = new SimpleStatBoost(STRENGTH, true, getMeleeEffect(client, STRENGTH));
+
+		changes.setStatChanges(Stream.of(
+			Stream.of(restoreChange),
+			Stream.of(defenceBoost.effect(client)),
+			Stream.of(OFFENSIVE_MELEE_STATS)
+				.map(stat ->
+				{
+					if (stat == ATTACK)
+					{
+						attackBoost.setStat(stat);
+						return attackBoost.effect(client);
+					}
+					if (stat == STRENGTH)
+					{
+						strengthBoost.setStat(stat);
+						return strengthBoost.effect(client);
+					}
+					return null;
+				}))
+			.reduce(Stream::concat)
+			.orElseGet(Stream::empty)
+			.toArray(StatChange[]::new));
+
+		changes.setPositivity(Stream.of(changes.getStatChanges())
+			.map(StatChange::getPositivity)
+			.max(Comparator.naturalOrder()).orElseThrow());
+		return changes;
+	}
+
+	private DeltaCalculator getMeleeEffect(Client client, Stat stat)
+	{
+		return Arrays.stream(MeleeEffect.values())
+			.filter(t -> t.getStat() == stat && client.getBoostedSkillLevel(Skill.HERBLORE) >= t.getLevelRequirement())
+			.max(Comparator.comparingInt(MeleeEffect::getLevelRequirement))
+			.orElseGet(() -> MeleeEffect.NONE).getDeltaCalculator();
+	}
+}
+
+@Getter
+enum MeleeEffect
+{
+	ATTACK_POTION(ATTACK, 3, perc(.10, 3)),
+	DEFENCE_POTION(DEFENCE, 30, perc(.10, 3)),
+	// Misnomer: Moonlight potion gives the effect of smelling salts on defence but requires the level for,
+	// and activates the var/effect timer for—albeit without the drain prevention from—Divine Super Defence potion
+	DIVINE_SUPER_DEFENCE_POTION(DEFENCE, 70, perc(.16, 11)),
+	STRENGTH_POTION(STRENGTH, 12, perc(.10, 3)),
+	SUPER_ATTACK_POTION(ATTACK, 45, perc(.15, 5)),
+	SUPER_DEFENCE_POTION(DEFENCE, 66, perc(.15, 5)),
+	SUPER_STRENGTH_POTION(STRENGTH, 55, perc(.15, 5)),
+	NONE(null, 0, max -> 0),
+	;
+
+	private final Stat stat;
+	private final int levelRequirement;
+	private final DeltaCalculator deltaCalculator;
```

Comment:
Most of these effects are already defined in ItemStatChanges. (some/all are even defined as variables for reuse in mixed potion definitions) If possible, I'd like to make those definitions static variables which can be referenced here instead of being duplicated.